source/LoadInsight.py

```diff
@@ -0,0 +1,85 @@
+
+# define the RBSA pipeline
+rbsa.add([
+    external.load(input="RBSA",output="config_RBSA")
+    extract.rbsa(input="config_RBSA", output="RBSA_data"),
+    clean.rbsa(input="RBSA_data", output="RBSA_loads"),
+    external.load(input="DEVICE_MAP", output="device_map"),
+    group.devices(input=["device_map","RBSA_loads"], output="site_loads"),
+    group.zipcodes(input="RBSA_data", output="zipcode_map"),
+    group.sites(input=["zipcode_map","site_loads"], output="area_loads"),
+    external.load(input="NOAA_DATA", output="NOAA_data"),
+    index.heat_cool(input="NOAA_DATA", output="heat_cool_index")
+    group.enduses(input=["heat_cool_index","area_loads"], output="enduse_loads"),
+    external.load(input="GAS_FRACTIONS", output="gas_fractions"),
+    electrification.undiscount_gas(input=["enduse_loads","gas_fractions"], output="total_loads"),
+    external.load(input="LOAD_NORMALIZATION_RULES", output="normalization_rules"),
+    normalize.max(input=["normalization_rules","total_loads"], output="normal_loads"),
+    sensitivity.find(input="normal_loads", output="loadshapes"),
+    external.load(input="TMY_DATA", output="tmy_data"),
+    project.loadshape(input=["tmy_data","loadshape"], output="total_loadshapes"),
+    electrification.discount_gas(input=["gas_fractions","total_loadshapes"], output="enduse_loadshapes"),
+    normalize.max(input=["normalization_rules","enduse_loadshapes"], output="normal_loadshapes"),
+    external.load(input="DAYTYPE_DEFINITIONS", output="daytype_definitions"),
+    extract.daytypes(input=["daytype_definitions","normal_loadshapes"], output="residential_composition"),
+    pipeline.send(input="residential_composition",output="ceus")
+])
+
+# create the CEUS pipeline
+ceus = pipeline.pipeline(name="ceus")
+
+# define the CEUS pipeline
+ceus.add([ # TODO
+    ])
+
+# create the feeder pipeline
+feeder = pipepline.pipeline(name="feeder")
+
+# define the feeder pipeline
+feeder.add([
+    external.rules(input="RULES_OF_ASSOCIATION",output="rules_of_association"),
+    pipeline.receive(input="rbsa",output="residential_composition"),
+    compose.rules(input=["residential_composition","rules_of_association"], output="residential_enduse_components"),
+    external.building_mix(input="RESIDENTIAL_MIX",output="residential_mix"),
+    compose.buildings(input=["residential_mix","residential_enduse_components"], output="residential_components"),
+    pipeline.receive(input="ceus",output="commercial_composition"),
+    compose.rules(input=["commercial_composition","rules_of_association"], output="commercial_enduse_components"),
+    external.building_mix(input="COMMERCIAL_MIX",output="commercial_mix"),
+    compose.buildings(input=["commercial_mix","commercial_enduse_components"], output="commercial_components"),
+    external.feeders(input="FEEDER_MIX", output="feeder_mix"),
+    compose.feeder(input=["commercial_components","residential_components","feeder_mix"], output="feeder_composition"),
```

Comment:
Inputs should always respect a single data type (in most cases), there are exceptions of course. I'd make them all arrays. 

Better yet, I think we can provide a dict (map) of input and outputs

Reply:
I haven't implemented the new format in the full LoadInsight.py yet.  I'm working on it right now.  The update is only for the example.

Reply:
We could just use kwargs.

Reply:
Actually we should be careful about the input and output args.  They are ordered because some of the functions may not commutative, etc.